cdap-watchdog/src/main/java/io/cdap/cdap/logging/clean/LogFolderCleaner.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.logging.clean;
+
+import org.apache.twill.filesystem.Location;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * LogFolderCleaner is responsible for cleaning up accumulated empty logs folders.
+ */
+public class LogFolderCleaner implements Runnable {
```

Comment:
Why implements Runnable? Is it scheduled or executed through executor?

Reply:
No need to implement Runnable now. Removed it.